### ISSUE TITLE
Socket server does not properly start after reload

### ISSUE THREAD

**Issue by jabbink** [jabbink/PokemonGoBot#780]
**Description:**
After stopping/starting/unloading/loading a bot the Socket server does not properly start

**Steps to reproduce:**

1. Start bot
2. stop it
3. start it

**Expected behavior:**
Socket server should be running again

**Actual behavior:**
Sometimes it works.

When reloading too fast, the port is still in use, when waiting and then starting again, the port is not in use, but it still doesn't start the server. Also in that last case, http://localhost:8080/api/bots returns `[]` even though the console still shows that it is walking/catching.

**Version:**
56adba1a79ae42c84c52838883e16c77134d20c6

**Operating System:**
Ubuntu 14.04

**Java version:**
```
java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)
```

**Comment by jozsi** [comment]
This seems to be a more general problem. I killed the app using `ctrl+c` but it didn't close the port.

**Comment by fabioneves** [comment]
Adding in, since it's related. If the bot can't find pokestops, it gets in a loop "Retrying in 10 seconds...", you can't stop or unload through the API.

**Comment by jabbink** [comment]
Decrease the initial size for that. Different issue though.

**Comment by casper-gh** [comment]
@fabioneves create a cronjob and check if the jar is running every 5 mins or less, if it does restart the bot, problem solved

**Comment by fabioneves** [comment]
@casper-gh i'm running it from the source. That's not really the issue, the thing is that the bot keeps trying to find pokestops, even I issue the unload or stop command from the API, it keeps going like that at least 10 times (I didn't wait longer).

Anyway, a different issue like jabbink said.

**Comment by jabbink** [comment]
https://github.com/jabbink/PokemonGoBot/issues/780#issuecomment-237279334

Still different issue. Stay on topic, please.

**Comment by langerhans** [comment]
Fixed with #950
